ksqldb-engine/src/main/java/io/confluent/ksql/util/PersistentQueryMetadata.java

```diff
@@ -114,6 +120,15 @@ protected PersistentQueryMetadata(
     this.materializationProvider = other.materializationProvider;
     this.physicalPlan = other.physicalPlan;
     this.materializationProviderBuilder = other.materializationProviderBuilder;
+    this.processingLogger = other.processingLogger;
+  }
+
+  @Override
+  protected void uncaughtHandler(final Thread thread, final Throwable error) {
+    super.uncaughtHandler(thread, error);
+
+    processingLogger.error(KafkaStreamsThreadError.of(
```

Comment:
consider logging this at the fatal level, since this is a thread-killing error.

Reply:
How do I log at fatal level?

The `ProcessingLogger` has only `error`, an the `inner` logger has only `error, info, debug`.